x-pack/plugins/code/public/monaco/monaco_helper.ts

```diff
@@ -43,6 +43,7 @@ export class MonacoHelper {
       this.monaco.languages.registerDefinitionProvider('java', definitionProvider);
       this.monaco.languages.registerDefinitionProvider('typescript', definitionProvider);
       this.monaco.languages.registerDefinitionProvider('javascript', definitionProvider);
+      this.monaco.languages.registerDefinitionProvider('go', definitionProvider);
```

Comment:
btw, do we have a global flag/option to safeguard the go support? @zfy0701 If so, it's better to include the flag in here.

Reply:
we need flag, let's make sure production mode there is no go language server shown

Reply:
> we need flag, let's make sure production mode there is no go language server shown

Any idea? For now, we only enable go language server under the development mode, I can't find a trivial way to carry the `serverOptions` or `env.dev` from `init()`to the monaco editor.

Reply:
I'll ask spencer

Reply:
Thanks for your tips, it just works under development mode now. But the `import` statement is not restricted to development mode because I can't find a proper conditional imports approach.

Reply:
it can be done by using if .. else and require 'xxx'

but I think this is totally fine now,

Reply:
can you check if language sever management item need the same check?

Reply:
There is no need for that. The langserver management is decided by the `LanguageServers: LanguageServerDefinition[]` array and this array is under the check, see [init.ts](https://github.com/elastic/kibana/blob/feature/merge-code/x-pack/plugins/code/server/init.ts#L144).

btw, what is the official way to run the kibana in production mode? I just test it in a hacky way, change the `env.dev` to `false` in the source code manually.